File: application/backend/business/invoice-v1/src/main/java/com/demo/bbq/business/invoice/domain/repository/database/product/ProductRepository.java
package com.demo.bbq.business.invoice.domain.repository.database.product;

import com.demo.bbq.business.invoice.domain.repository.database.product.entity.ProductEntity;
import org.springframework.data.repository.CrudRepository;

public interface ProductRepository extends CrudRepository<ProductEntity, Long> {
}

File: backend/business/product-v1/cmd/products/repository/product_repository.go
package repository

import (
	"com.demo.poc/cmd/products/repository/entity"
)

type ProductRepository interface {
	FindAll() ([]entity.ProductEntity, error)

	FindByScope(scope string) ([]entity.ProductEntity, error)

	FindByCode(code string) (*entity.ProductEntity, error)

	Save(product *entity.ProductEntity) error

	Delete(code string) error
}
